real-esrgan/real-esrgan.cpp

```diff
@@ -117,7 +118,7 @@ ICudaEngine* build_engine(unsigned int maxBatchSize, IBuilder* builder, IBuilder
     std::cout << "Build engine successfully!" << std::endl;
 
     // Don't need the network any more
-    delete network;
+    network->destroy();
```

Comment:
Why do we use destroy()?  Seems destroy is deprecated in TRT8.

Reply:
1. I found that if delete is used, an error will be reported.
```
real-esrgan.cpp:120:12: error: _virtual nvinfer1::INetworkDefinition::~INetworkDefinition()_ is protected within this context
  120 |     delete network;
      |            ^~~~~~~
compilation terminated due to -Wfatal-errors
```
2. In addition, I found that in yolov3, yolov4, and yolov5 of tensorRTx, destroy is used instead of delete.